aspnetcore/migration/22-to-30.md

```diff
@@ -725,6 +725,39 @@ public class HomeController : ControllerBase
 }
 ```
 
+Authorization can also be configured for specific classes of endpoints. The following code is an example of converting an MVC app that configured a global `AuthorizeFilter` to an app with a specific policy to use `RequireAuthorization`:
+
+```csharp
+public void ConfigureServices(IServiceCollection services)
+{
+    ...
+    // Pre 3.0:
+    // services.AddMvc(options => options.Filters.Add(new AuthorizeFilter(...));
+    services.AddControllers();
+}
+
+public void Configure(IApplicationBuilder app)
+{
+    
+    ...
+    app.UseRouting();
+    app.UseAuthentication();
+    app.UseAuthorization();
+
+    app.UseEndpoints(endpoints =>
+    {
+        // using Microsoft.AspNetCore.Mvc.Authorization;
+        // using Microsoft.AspNetCore.Authorization;
+        var policy = new AuthorizationPolicyBuilder()
+                         .RequireAuthenticatedUser()
+                         .Build();
+        
+        endpoints.MapControllers()
+            .RequireAuthorization(policy);
```

Comment:
I can't find an overload of `RequireAuthorization` that takes an `AuthorizationPolicy`. Is there an additional package needed for this?

Reply:
You're right. This doesn't exist. You have to add policies by name which is a bit unfortunate. This would look like:

```C#
services.AddAuthorization(o => o.AddPolicy("RequireAuthenticatedUserPolicy", builder => builder.RequireAuthenticatedUser()));

...

endpoints.MapControllers().RequireAuthorization("RequireAuthenticatedUserPolicy");
```

Reply:
Yeah, that makes sense. I wondered if I just couldn't see this overload because it was in a NuGet package somewhere. I couldn't find anything obvious through searching so I asked the question here.